--- src/utils/vk_config.cpp.orig	2021-05-18 05:20:37.000000000 -0400
+++ src/utils/vk_config.cpp	2021-05-18 05:22:27.000000000 -0400
@@ -25,6 +25,12 @@
 #include "utils/vk_utils.h"
 
 #include <pwd.h>     // getpwuid
+#if defined(HAVE_SYS_TYPES_H) || defined(__APPLE__)
+# include <sys/types.h>
+#endif /* HAVE_SYS_TYPES_H || __APPLE__ */
+#if defined(HAVE_UNISTD_H) || defined(__APPLE__)
+# include <unistd.h>
+#endif /* HAVE_UNISTD_H || __APPLE__ */
 
 #include <QColor>
 #include <QDir>
