--- buildlib/environment.mak.in.orig	2004-05-08 15:41:01.000000000 -0400
+++ buildlib/environment.mak.in	2014-04-09 17:23:00.000000000 -0400
@@ -62,7 +62,19 @@
 ifneq ($(words $(filter linux-gnu gnu% %gnu,$(HOST_OS))),0)
    SONAME_MAGIC=-Wl,-soname -Wl,
    LFLAGS_SO=
-else
+endif
+
+ifneq ($(words $(filter freebsd% %freebsd,$(HOST_OS))),0)
+   SONAME_MAGIC=-Wl,-soname -Wl,
+   LFLAGS_SO=
+endif
+
+ifneq ($(words $(filter darwin% %darwin,$(HOST_OS))),0)
+   SONAME_MAGIC=-install_name /opt/local/lib/
+   LFLAGS_SO=-dynamiclib
+endif
+
+ifndef SONAME_MAGIC
    # Do not know how to create shared libraries here.
    ONLYSTATICLIBS = yes
 endif
