--- apt-pkg/deb/deblistparser.cc.orig	2003-09-22 00:16:26.000000000 -0400
+++ apt-pkg/deb/deblistparser.cc	2014-04-10 12:01:31.000000000 -0400
@@ -163,18 +163,18 @@
       /* Strip out any spaces from the text, this undoes dpkgs reformatting
          of certain fields. dpkg also has the rather interesting notion of
          reformatting depends operators < -> <= */
-      char *I = S;
+      char *J = S;
       for (; Start != End; Start++)
       {
 	 if (isspace(*Start) == 0)
-	    *I++ = tolower(*Start);
+	    *J++ = tolower(*Start);
 	 if (*Start == '<' && Start[1] != '<' && Start[1] != '=')
-	    *I++ = '=';
+	    *J++ = '=';
 	 if (*Start == '>' && Start[1] != '>' && Start[1] != '=')
-	    *I++ = '=';
+	    *J++ = '=';
       }
 
-      Result = AddCRC16(Result,S,I - S);
+      Result = AddCRC16(Result, S, (J - S));
    }
    
    return Result;
