--- apt-pkg/deb/dpkgpm.cc.orig	2014-04-09 17:22:59.000000000 -0400
+++ apt-pkg/deb/dpkgpm.cc	2014-04-09 17:22:59.000000000 -0400
@@ -436,8 +436,8 @@
          it forks scripts. What happens is that when you hit ctrl-c it sends
 	 it to all processes in the group. Since dpkg ignores the signal 
 	 it doesn't die but we do! So we must also ignore it */
-      sighandler_t old_SIGQUIT = signal(SIGQUIT,SIG_IGN);
-      sighandler_t old_SIGINT = signal(SIGINT,SIG_IGN);
+      sig_t old_SIGQUIT = signal(SIGQUIT,SIG_IGN);
+      sig_t old_SIGINT = signal(SIGINT,SIG_IGN);
 		     
       // Fork dpkg
       pid_t Child = ExecFork();
