--- configure.in	2012-03-27 08:15:03.000000000 -0400
+++ configure.ac	2013-05-05 00:40:31.000000000 -0400
@@ -1,5 +1,5 @@
-AC_PREREQ(2.59)
-AC_REVISION($Id$)
+AC_PREREQ([2.59])
+AC_REVISION([$Id$])
 #
 #  configure.in
 #
@@ -85,10 +85,10 @@
 	[iodbc@openlinksw.com],
 	[libiodbc])
 AC_CONFIG_SRCDIR([include/isql.h])
-AC_CONFIG_AUX_DIR(admin)
-AC_CONFIG_MACRO_DIR(admin)
-AM_CONFIG_HEADER(include/config.h)
-AM_INIT_AUTOMAKE(AC_PACKAGE_TARNAME, AC_PACKAGE_VERSION)
+AC_CONFIG_AUX_DIR([admin])
+AC_CONFIG_MACRO_DIR([admin])
+AC_CONFIG_HEADERS([include/config.h])
+AM_INIT_AUTOMAKE([foreign])
 AC_CANONICAL_HOST
 AM_MAINTAINER_MODE
 
@@ -120,7 +120,7 @@
 ##########################################################################
 AC_MSG_CHECKING(cached information)
 hostcheck="$host"
-AC_CACHE_VAL(ac_cv_hostcheck, [ ac_cv_hostcheck="$hostcheck" ])
+AC_CACHE_VAL([ac_cv_hostcheck], [ ac_cv_hostcheck="$hostcheck" ])
 if test "$ac_cv_hostcheck" != "$hostcheck"; then
   AC_MSG_RESULT(changed)
 
@@ -160,10 +160,9 @@
 ##  Set Layout for iODBC						##
 ##									##
 ##########################################################################
-AC_PREFIX_DEFAULT(/usr/local)
 
 AC_MSG_CHECKING(for iODBC installation layout)
-AC_ARG_WITH(layout, [dnl
+AC_ARG_WITH([layout], [dnl
 AC_HELP_STRING([--with-layout=LAYOUT], [Choose installation layout])dnl
 ],
 [ case "${withval}" in
@@ -273,18 +272,27 @@
 ##########################################################################
 AC_HEADER_STDC
 AC_HEADER_TIME
-AC_CHECK_HEADERS(wchar.h)
+AC_CHECK_HEADERS([wchar.h])
+AC_CHECK_HEADERS([fcntl.h locale.h nlist.h sys/param.h sys/time.h])
 
+# Checks for typedefs, structures, and compiler characteristics.
+AC_CHECK_HEADER_STDBOOL
+AC_TYPE_SIZE_T
+AC_TYPE_SSIZE_T
 
 ##########################################################################
 ##									##
 ##  Check for specific library functions				##
 ##									##
 ##########################################################################
-AC_CHECK_FUNCS(strerror setenv asprintf snprintf gettimeofday localtime_r)
+AC_FUNC_MALLOC
+AC_FUNC_REALLOC
+AC_CHECK_FUNCS([memmove memset mkdir putenv setlocale strcasecmp strchr strdup strncasecmp strrchr strstr tzset])
+
+AC_CHECK_FUNCS([strerror setenv asprintf snprintf gettimeofday localtime_r])
 
 dnl Unicode
-AC_CHECK_FUNCS(wcslen wcscpy wcsncpy wcschr wcscat wcscmp towlower wcsncasecmp)
+AC_CHECK_FUNCS([wcslen wcscpy wcsncpy wcschr wcscat wcscmp towlower wcsncasecmp])
 
 
 ##########################################################################
@@ -292,7 +300,7 @@
 ##  Check if we want to build the GUI applications and libraries	##
 ##									##
 ##########################################################################
-AC_ARG_ENABLE(gui, [dnl
+AC_ARG_ENABLE([gui], [dnl
 AC_HELP_STRING([--enable-gui], [build GUI applications (default)])
 AC_HELP_STRING([--disable-gui], [build GUI applications])dnl
 ],
@@ -352,34 +360,34 @@
 ##									##
 ##########################################################################
 
-AC_CHECK_HEADERS(dlfcn.h dl.h dld.h)
+AC_CHECK_HEADERS([dlfcn.h dl.h dld.h])
 
 LIBADD_DL=""
 
-AC_CHECK_LIB(dl, dlopen, 
-  [AC_DEFINE(HAVE_LIBDL, 1, [Define if you have the -ldl library]) 
+AC_CHECK_LIB([dl], [dlopen], 
+  [AC_DEFINE([HAVE_LIBDL], [1], [Define if you have the -ldl library]) 
 	LIBADD_DL="-ldl"],
-  [AC_CHECK_FUNCS(dlopen _dlopen, [
-	AC_DEFINE(HAVE_LIBDL, 1,[Define if you have the -ldl library])
+  [AC_CHECK_FUNCS([dlopen _dlopen], [
+	AC_DEFINE([HAVE_LIBDL], [1],[Define if you have the -ldl library])
   ])]
   )
 
-AC_CHECK_FUNC(shl_load, 
-  [AC_DEFINE(HAVE_SHL_LOAD, 1, [Define if you have the shl_load function])],
-  [AC_CHECK_LIB(dld, shl_load, 
-    [AC_DEFINE(HAVE_SHL_LOAD, 1, [Define if you have the shl_load function]) LIBADD_DL="$LIBADD_DL -ldld"])]
+AC_CHECK_FUNC([shl_load], 
+  [AC_DEFINE([HAVE_SHL_LOAD], [1], [Define if you have the shl_load function])],
+  [AC_CHECK_LIB([dld], [shl_load], 
+    [AC_DEFINE([HAVE_SHL_LOAD], [1], [Define if you have the shl_load function]) LIBADD_DL="$LIBADD_DL -ldld"])]
   )
 
-AC_CHECK_LIB(dld, dld_link, 
-   [AC_DEFINE(HAVE_DLD, 1, [Define if you have the -ldld library])dnl
+AC_CHECK_LIB([dld], [dld_link], 
+   [AC_DEFINE([HAVE_DLD], [1], [Define if you have the -ldld library])dnl
     test "x$ac_cv_lib_dld_shl_load" = yes || LIBADD_DL="$LIBADD_DL -ldld"])
 
-AC_CHECK_FUNC(NSCreateObjectFileImageFromFile,
-    [AC_DEFINE(HAVE_DYLD, 1, [Define if you have the DYLD library])])
+AC_CHECK_FUNC([NSCreateObjectFileImageFromFile],
+    [AC_DEFINE([HAVE_DYLD], [1], [Define if you have the DYLD library])])
 
 AC_SUBST(LIBADD_DL)
 
-AC_CACHE_CHECK([for underscore before symbols], libltdl_cv_uscore, [
+AC_CACHE_CHECK([for underscore before symbols], [libltdl_cv_uscore], [
   echo "int main(){int i=1; return 0;} void fnord(){int i=23; int ltuae=42;}" > conftest.c
   ${CC} -c conftest.c > /dev/null
   if (nm conftest.o | grep _fnord) > /dev/null; then
@@ -464,7 +472,7 @@
 fi
 
 if test x"$libltdl_cv_need_uscore" = xyes; then
-   AC_DEFINE(NEED_USCORE, 1,
+   AC_DEFINE([NEED_USCORE], [1],
 	[Define if dlsym() requires a leading underscore in symbol names. ])
 fi
 
@@ -474,8 +482,8 @@
 ##  Check if the dl library has the dladdr function                     ##
 ##                                                                      ##
 ##########################################################################
-AC_CHECK_FUNC(dladdr, AC_DEFINE(HAVE_DLADDR, 1, [Define to 1 if you have the `dladdr' function.]),
-[AC_CHECK_LIB(dl, dladdr, [AC_DEFINE(HAVE_DLADDR, 1, [Define to 1 if you have the `dladdr' function]) LIBADD_DL="-ldl"],[])]
+AC_CHECK_FUNC([dladdr], [AC_DEFINE([HAVE_DLADDR], [1], [Define to 1 if you have the `dladdr' function.])],
+[AC_CHECK_LIB([dl], [dladdr], [AC_DEFINE([HAVE_DLADDR], [1], [Define to 1 if you have the `dladdr' function]) LIBADD_DL="-ldl"],[])]
 )
  
 AC_CHECK_TYPE([Dl_info],
@@ -492,7 +500,7 @@
 ##########################################################################
 
 AC_MSG_CHECKING(for iODBC mode)
-AC_ARG_ENABLE(odbc3, [dnl
+AC_ARG_ENABLE([odbc3], [dnl
 AC_HELP_STRING([--enable-odbc3], [build ODBC 3.x compatible driver manager (default)])
 AC_HELP_STRING([--disable-odbc3], [build ODBC 2.x compatible driver manager])dnl
 ],
@@ -552,7 +560,7 @@
 #  Now allow to overrule this directory with a custom setting
 #
 AC_MSG_CHECKING(for iODBC ini directory)
-AC_ARG_WITH(iodbc-inidir, [dnl
+AC_ARG_WITH([iodbc-inidir], [dnl
 AC_HELP_STRING([--with-iodbc-inidir=DIR], [where the system odbc.ini file should be located])
 AC_HELP_STRING([ ], [(default is /etc)])dnl
 ],
@@ -607,7 +615,7 @@
 ##########################################################################
 
 AC_MSG_CHECKING(for libodbc.so link)
-AC_ARG_ENABLE(libodbc, [dnl
+AC_ARG_ENABLE([libodbc], [dnl
 AC_HELP_STRING([--enable-libodbc], [install extra libodbc.so library (default)])
 AC_HELP_STRING([--disable-libodbc], [do not install extra libodbc.so library ])dnl
 ],
@@ -619,7 +627,7 @@
 ],[]) 
 AC_MSG_RESULT([$install_libodbc])
 
-AM_CONDITIONAL(INSTALL_LIBODBC, [test x$install_libodbc = xtrue])
+AM_CONDITIONAL([INSTALL_LIBODBC], [test x$install_libodbc = xtrue])
 
 
 ##########################################################################
@@ -629,27 +637,27 @@
 ##########################################################################
 
 # for backward compatibility
-AC_ARG_WITH(pthreads,[],
+AC_ARG_WITH([pthreads],[],
 [ 
   AC_MSG_WARN([--with-pthreads flag is deprecated, use --enable-pthreads])
 ],[]) 
 
 
 AC_MSG_CHECKING(for thread model)
-AC_ARG_ENABLE(pthreads, [dnl
+AC_ARG_ENABLE([pthreads], [dnl
 AC_HELP_STRING([--enable-pthreads], [build threaded libraries and applications (default)])
 AC_HELP_STRING([--disable-pthreads], [build non-threaded libraries and applications])dnl
 ],
 [ case "${enableval}" in
     yes) USE_PTHREADS=true ;;
     no)  USE_PTHREADS=false ;;
-    *) AC_MSG_ERROR(bad value ${enableval} for --enable-pthreads) ;;
+    *) AC_MSG_ERROR([bad value ${enableval} for --enable-pthreads]) ;;
   esac
 ],[USE_PTHREADS=true]) 
 
 if test x$USE_PTHREADS = xtrue
 then
-    AC_MSG_RESULT(enabled pthread support)
+    AC_MSG_RESULT([enabled pthread support])
     CPPFLAGS="$CPPFLAGS -DWITH_PTHREADS -D_REENTRANT"
     found_lib=false
 
@@ -704,20 +712,20 @@
     #
     #  Check headers
     #
-    AC_CHECK_HEADER(pthread.h)
+    AC_CHECK_HEADER([pthread.h])
 
     #
     #  If mutex functions are in C library, we should not need to link
     #  with -lpthread
     #
-    AC_CHECK_FUNC(pthread_mutex_lock, [found_lib=true])
+    AC_CHECK_FUNC([pthread_mutex_lock], [found_lib=true])
 
     #
     #  Check out the various libraries
     #
     if test x$found_lib = xfalse
     then
-	IODBC_CHECK_PTHREAD_LIB(pthreads, pthread_mutex_lock,
+	IODBC_CHECK_PTHREAD_LIB([pthreads], [pthread_mutex_lock],
 	[
 	    LIBS="$LIBS -lpthreads"
 	    found_lib=true
@@ -725,7 +733,7 @@
     fi
     if test x$found_lib = xfalse
     then
-	IODBC_CHECK_PTHREAD_LIB(pthread, pthread_mutex_lock, 
+	IODBC_CHECK_PTHREAD_LIB([pthread], [pthread_mutex_lock], 
 	[
 	    LIBS="$LIBS -lpthread"
 	    found_lib=true
@@ -743,7 +751,7 @@
 ##########################################################################
 LIBADD_OS=""
 AC_MSG_CHECKING(for OS dependent libraries and link flags)
-AM_CONDITIONAL(DARWIN, false)	
+AM_CONDITIONAL([DARWIN], [false])	
 case $host in
   *-hpux*)
 	LDFLAGS="$LDFLAGS -Wl,+s"
