--- libnessus/Makefile.orig	2014-07-13 19:20:54.000000000 -0400
+++ libnessus/Makefile	2014-07-13 19:35:33.000000000 -0400
@@ -3,13 +3,13 @@
 OBJS = plugutils.o system.o network.o resolve.o arglists.o ftp_funcs.o \
 	scanners_utils.o pcap.o hlst.o  harglists.o \
         getopt.o getopt1.o www_funcs.o diff.o ids_send.o rand.o popen.o \
-	bpf_share.o rand.o popen.o services.o services1.o \
+	rand.o popen.o services.o services1.o \
 	proctitle.o store.o snprintf.o kb.o share_fd.o
 	
 LO_OBJS = plugutils.lo system.lo network.lo resolve.lo arglists.lo \
 	ftp_funcs.lo scanners_utils.lo pcap.lo hlst.lo  harglists.lo \
 	getopt.lo getopt1.lo www_funcs.lo diff.lo ids_send.lo \
-	rand.lo popen.lo bpf_share.lo services.lo services1.lo \
+	rand.lo popen.lo services.lo services1.lo \
 	proctitle.lo store.lo snprintf.lo kb.lo share_fd.lo
 
 # += does not work with non-GNU make ...
