--- src/plugins/Makefile.am.orig	2013-10-06 02:58:14.000000000 -0400
+++ src/plugins/Makefile.am	2014-10-10 16:01:57.000000000 -0400
@@ -113,8 +113,8 @@
 # install plugin header
 
 install-data-hook:
-	$(INSTALL) -d $(includedir)/$(PACKAGE)
-	$(INSTALL_DATA) $(abs_top_srcdir)/src/plugins/weechat-plugin.h $(includedir)/$(PACKAGE)
+	$(INSTALL) -d $(DESTDIR)$(includedir)/$(PACKAGE)
+	$(INSTALL_DATA) $(abs_top_srcdir)/src/plugins/weechat-plugin.h $(DESTDIR)$(includedir)/$(PACKAGE)
 
 # uninstall plugin header
 
