--- Samples/Demo8/Makefile.am.orig	2013-01-12 12:07:24.000000000 -0500
+++ Samples/Demo8/Makefile.am	2014-06-19 13:36:30.000000000 -0400
@@ -1,10 +1,16 @@
-INCLUDES = -I$(top_srcdir)/cegui/include \
+## automake Makefile for cegui Demo8
+
+AM_CPPFLAGS = -I$(top_srcdir)/cegui/include \
            -I$(top_srcdir)/Samples/common/include \
            -I$(top_srcdir)/cegui/include/ScriptingModules/LuaScriptModule
 
-noinst_PROGRAMS = Demo8@cegui_bsfx@
+examplesdir = ${datarootdir}/examples/${PACKAGE_TARNAME}
+examples_bindir = ${examplesdir}/bin
+examples_includedir = ${examplesdir}/include
+
+examples_bin_PROGRAMS = Demo8@cegui_bsfx@
 
-noinst_HEADERS =  Sample_Demo8.h
+examples_include_HEADERS = Sample_Demo8.h
 
 if CEGUI_BUILD_STATIC
 Demo8@cegui_bsfx@_CPPFLAGS = -DCEGUI_STATIC
@@ -28,4 +34,4 @@
 
 
 install-exec-local:
-	$(LN_S) -f `pwd`/.libs/Demo8@cegui_bsfx@$(EXEEXT) $(top_builddir)/bin/Demo8@cegui_bsfx@$(EXEEXT)
+	$(LN_S) -fv `pwd`/.libs/Demo8@cegui_bsfx@$(EXEEXT) $(top_builddir)/bin/Demo8@cegui_bsfx@$(EXEEXT)
