--- config/unlocked-io.m4.orig	2013-08-15 13:56:17.000000000 -0400
+++ config/unlocked-io.m4	2014-03-13 09:58:46.000000000 -0400
@@ -6,13 +6,13 @@
 # gives unlimited permission to copy and/or distribute it,
 # with or without modifications, as long as this notice is preserved.
 
-dnl From Jim Meyering.
-dnl
-dnl See if the glibc *_unlocked I/O macros or functions are available.
-dnl Use only those *_unlocked macros or functions that are declared
-dnl (because some of them were declared in Solaris 2.5.1 but were removed
-dnl in Solaris 2.6, whereas we want binaries built on Solaris 2.5.1 to run
-dnl on Solaris 2.6).
+dnl# From Jim Meyering.
+dnl#
+dnl# See if the glibc *_unlocked I/O macros or functions are available.
+dnl# Use only those *_unlocked macros or functions that are declared
+dnl# (because some of them were declared in Solaris 2.5.1 but were removed
+dnl# in Solaris 2.6, whereas we want binaries built on Solaris 2.5.1 to run
+dnl# on Solaris 2.6).
 
 AC_DEFUN([gl_FUNC_GLIBC_UNLOCKED_IO],
 [
@@ -21,8 +21,8 @@
      Unlocked I/O can improve performance in unithreaded apps,
      but it is not safe for multithreaded apps.])
 
-  dnl Persuade glibc and Solaris <stdio.h> to declare
-  dnl fgets_unlocked(), fputs_unlocked() etc.
+  dnl# Persuade glibc and Solaris <stdio.h> to declare
+  dnl# fgets_unlocked(), fputs_unlocked() etc.
   AC_REQUIRE([gl_USE_SYSTEM_EXTENSIONS])
 
   AC_CHECK_DECLS_ONCE([clearerr_unlocked])
