--- doc/Makefile.am.orig	2013-08-15 13:56:17.000000000 -0400
+++ doc/Makefile.am	2014-03-13 09:58:46.000000000 -0400
@@ -22,9 +22,10 @@
 ## ---------------------------------------------------------------------
 
 # This variable is needed to subvert automake's info rules.
-# They don't work for generated texi files:
+# They do NOT work for generated texi files:
 #
 INFO_DEPS       = autogen.info
+# (ignore the warning, it says above we are overriding the rule on purpose)
 MIexe           = $(MAKEINFO) --no-split
 MAKEINFOFLAGS   = -I$(top_srcdir)/autoopts -I../autoopts
 passenv         = MAKE=$(MAKE) srcdir="$(srcdir)" SHELL="$(POSIX_SHELL)" \
@@ -50,7 +51,8 @@
 agdoc.texi      : # self-depends upon all executables
 	$(run_mktexi) $@
 
-autogen.dvi     : agdoc.texi
+dependency_for_autogen.dvi: agdoc.texi
+.PHONY: dependency_for_autogen.dvi
 autogen.texi    : agdoc.texi mk-agen-texi.sh
 
 # Special rule for generating all the GNU standard formats.
