--- configure.ac.bak	2013-08-09 12:22:32.000000000 -0400
+++ configure.ac	2013-08-09 12:27:03.000000000 -0400
@@ -1,19 +1,83 @@
-# To compile this into a configure script, you need:
-# * Autoconf 2.50 or newer
-# * Gtk (for $prefix/share/aclocal/gtk.m4)
-# * Automake (for aclocal)
-# If you've got them, running "autoreconf" should work.
-
-# Version number is substituted by Buildscr for releases, snapshots
-# and custom builds out of svn; X.XX shows up in ad-hoc developer
-# builds, which shouldn't matter
-AC_INIT(putty, 0.63)
+dnl#                                               -*- Autoconf -*-
+dnl# To compile this into a configure script, you need:
+dnl# * Autoconf 2.50 or newer
+dnl# * Gtk (for $prefix/share/aclocal/gtk.m4)
+dnl# * Automake (for aclocal)
+dnl# If you've got them, running "autoreconf" should work.
+
+dnl# Version number is substituted by Buildscr for releases, snapshots
+dnl# and custom builds out of svn; X.XX shows up in ad-hoc developer
+dnl# builds, which shouldn't matter
+AC_INIT([putty],[0.63])
 AC_CONFIG_FILES([Makefile])
 AC_CONFIG_HEADERS([uxconfig.h:uxconfig.in])
-AM_INIT_AUTOMAKE([-Wall -Werror foreign])
+AC_CANONICAL_TARGET
+AC_USE_SYSTEM_EXTENSIONS
+AM_INIT_AUTOMAKE([-Wall subdir-objects foreign dist-zip])
+AM_MAINTAINER_MODE
+AC_OBJEXT
+AC_EXEEXT
+
+# Checks for programs.
+AM_PROG_AR
+if test "x$CC" = "x"; then
+	test -z "$CC"
+	AC_PROG_CC
+else
+	test ! -z "$CC"
+	AC_SUBST([CC])
+fi
+if test "x$CPP" = "x"; then
+	test -z "$CPP"
+	AC_PROG_CPP
+else
+	test ! -z "$CPP"
+	AC_SUBST([CPP])
+fi
+if test "x$INSTALL" = "x"; then
+	test -z "$INSTALL"
+	AC_PROG_INSTALL
+else
+	test ! -z "$INSTALL"
+	AC_SUBST([INSTALL])
+fi
+if test "x$LN_S" = "x"; then
+	test -z "$LN_S"
+	AC_PROG_LN_S
+else
+	test ! -z "$LN_S"
+	AC_SUBST([LN_S])
+fi
+if test "x$RANLIB" = "x"; then
+	test -z "$RANLIB"
+	AC_PROG_RANLIB
+else
+	test ! -z "$RANLIB"
+	AC_SUBST([RANLIB])
+fi
 
-AC_PROG_INSTALL
-AC_PROG_RANLIB
+# Checks for libraries.
+# FIXME: Replace `main' with a function in `-lgtk':
+AC_CHECK_LIB([gtk], [main])
+
+# Checks for header files.
+AC_PATH_X
+AC_PATH_XTRA
+AC_CHECK_HEADERS([arpa/inet.h fcntl.h limits.h locale.h netdb.h netinet/in.h stdio.h sys/ioctl.h sys/socket.h sys/time.h termios.h utime.h utmp.h wchar.h])
+
+# Checks for typedefs, structures, and compiler characteristics.
+AC_TYPE_UID_T
+AC_TYPE_MODE_T
+AC_TYPE_OFF_T
+AC_TYPE_PID_T
+AC_TYPE_SIZE_T
+AC_TYPE_UINT32_T
+
+# Checks for library functions.
+AC_FUNC_CHOWN
+AC_FUNC_FORK
+AC_FUNC_MBRTOWC
+AC_CHECK_FUNCS([atexit clock_gettime dup2 endpwent getcwd getwd gethostbyname gethostname gettimeofday inet_ntoa memchr memmove memset mkdir putenv select setlocale socket strcasecmp strchr strcspn strerror strrchr strspn strstr utime])
 
 # Mild abuse of the '--enable' option format to allow manual
 # specification of setuid or setgid setup in pterm.
@@ -32,13 +96,13 @@
                no) setidtype=none;;
                *) setidtype=setgid; setidval="$enableval";;
              esac])
-AM_CONDITIONAL(HAVE_SETID_CMD, [test "$setidtype" != "none"])
+AM_CONDITIONAL([HAVE_SETID_CMD], [test "$setidtype" != "none"])
 AS_IF([test "x$setidtype" = "xsetuid"],
       [SETID_CMD="chown $setidval"; SETID_MODE="4755"])
 AS_IF([test "x$setidtype" = "xsetgid"],
       [SETID_CMD="chgrp $setidval"; SETID_MODE="2755"])
-AC_SUBST(SETID_CMD)
-AC_SUBST(SETID_MODE)
+AC_SUBST([SETID_CMD])
+AC_SUBST([SETID_MODE])
 
 AC_ARG_WITH([gssapi],
   [AS_HELP_STRING([--without-gssapi],
@@ -61,10 +125,10 @@
 case "$gtk_version_desired" in
   1 | 2 | any | no) ;;
   yes) gtk_version_desired="any" ;;
-  *) AC_ERROR([Invalid GTK version specified])
+  *) AC_MSG_ERROR([Invalid GTK version specified])
 esac
 
-AC_CHECK_HEADERS([utmpx.h sys/select.h],,,[
+AC_CHECK_HEADERS([utmpx.h sys/select.h],[],[],[
 #include <sys/types.h>
 #include <utmp.h>])
 
@@ -87,7 +151,7 @@
     AM_PATH_GTK([1.2.0], [gtk=1], [])
     ],[
     # manual check for gtk1
-    AC_PATH_PROG(GTK1_CONFIG, gtk-config, absent)
+    AC_PATH_PROG([GTK1_CONFIG], [gtk-config], [absent])
     if test "$GTK1_CONFIG" != "absent"; then
       GTK_CFLAGS=`"$GTK1_CONFIG" --cflags`
       GTK_LIBS=`"$GTK1_CONFIG" --libs`
@@ -97,7 +161,9 @@
     ;;
 esac
 
-AM_CONDITIONAL(HAVE_GTK, [test "$gtk" != "none"])
+GTK_DOC_CHECK
+
+AM_CONDITIONAL([HAVE_GTK], [test "$gtk" != "none"])
 
 if test "$gtk" = "2"; then
   ac_save_CFLAGS="$CFLAGS"
@@ -112,17 +178,17 @@
 AC_SEARCH_LIBS([socket], [xnet])
 
 AS_IF([test "x$with_gssapi" != xno],
-  [AC_SEARCH_LIBS(
-    [dlopen],[dl],
+  [AC_SEARCH_LIBS([dlopen],
+    [dl],
     [],
     [AC_DEFINE([NO_LIBDL], [1], [Define if we could not find libdl.])
      AC_CHECK_HEADERS([gssapi/gssapi.h])
-     AC_SEARCH_LIBS(
-       [gss_init_sec_context],[gssapi gssapi_krb5 gss],
+     AC_SEARCH_LIBS([gss_init_sec_context],
+       [gssapi gssapi_krb5 gss],
        [],
        [AC_DEFINE([NO_GSSAPI_LIB], [1], [Define if we could not find a gssapi library])])])])
 
-AC_CHECK_LIB(X11, XOpenDisplay,
+AC_CHECK_LIB([X11], [XOpenDisplay],
              [GTK_LIBS="-lX11 $GTK_LIBS"
               AC_DEFINE([HAVE_LIBX11],[],[Define if libX11.a is available])])
 
@@ -132,10 +198,10 @@
 
 if test "x$GCC" = "xyes"; then
   :
-  AC_SUBST(WARNINGOPTS, ['-Wall -Werror'])
+  AC_SUBST([WARNINGOPTS], ['-Wall'])
 else
   :
-  AC_SUBST(WARNINGOPTS, [])
+  AC_SUBST([WARNINGOPTS], [])
 fi
 
 AC_OUTPUT
