--- /opt/local/var/macports/sources/rsync.macports.org/release/tarballs/ports/sysutils/macportsscripts/Portfile	2013-03-20 13:30:16.000000000 -0400
+++ /opt/local/var/macports/sources/LocalPorts/sysutils/macportsscripts/Portfile	2013-05-15 16:52:50.000000000 -0400
@@ -5,7 +5,7 @@
 PortGroup       github 1.0
 
 epoch           2
-github.setup    cooljeanius macportsscripts 0.2.0 v
+github.setup    cooljeanius macportsscripts 0.3.0 v
 categories      sysutils macports
 platforms       darwin
 maintainers     phw gwmail.gwu.edu:egall openmaintainer
@@ -16,8 +16,8 @@
 
 long_description ${description}
 
-checksums       rmd160 3d24d9fdf6dacfab405e9ec7772e7969e0771270 \
-                sha256 62261afafd9021876c9e2717e76165e365ef71f9936ada392ee89ed561fc80ca
+checksums       rmd160  bfa65ac2a6da30474d76e6c89e1ad8425695173b \
+                sha256  b067846c2d4e5c122c51647a59ff9b1f93f2c9d4b3e32173e6d9a933cabbe4ea
 
 use_configure   no
 build {}
@@ -25,4 +25,6 @@
 destroot {
     eval xinstall -m 755 [glob ${worksrcpath}/*.sh] ${destroot}${prefix}/bin
     eval xinstall -m 755 [glob ${worksrcpath}/*.pl] ${destroot}${prefix}/bin
+    xinstall -d ${destroot}${prefix}/share/doc/${name}
+    eval xinstall -m 644 [glob ${worksrcpath}/*.md] ${destroot}${prefix}/share/doc/${name}
 }
