--- Portfile.orig.2	2013-01-29 04:17:54.000000000 -0500
+++ Portfile	2014-07-30 12:40:03.000000000 -0400
@@ -5,40 +5,48 @@
 
 name                cpan2port
 version             0.1.0
-categories          sysutils macports
+categories          sysutils macports perl
 platforms           darwin
 maintainers         gwmail.gwu.edu:egall openmaintainer
 license             WTFPL-2 permissive
 supported_archs     noarch
 
 description         A perl script to generate MacPorts portfiles
-long_description    ${description} from CPAN modules
-homepage            http://svn.macports.org/repository/macports/contrib/cpan2port
+long_description    ${description} from CPAN modules.
+homepage            http://svn.macports.org/repository/macports/contrib/${name}
 
 livecheck.type      none
 
 fetch.type          svn
-svn.url             https://svn.macports.org/repository/macports/contrib/cpan2port
-svn.revision        101883
-
-depends_lib-append  port:p5.12-module-depends \
-                    port:p5.12-list-moreutils \
-                    port:p5.12-module-corelist \
-                    port:p5.12-yaml \
-                    port:p5.12-file-path \
-                    port:p5.12-pod-simple \
-                    port:p5.12-carp-clan \
-                    port:p5.12-getopt-simple
+svn.url             https://svn.macports.org/repository/macports/contrib/${name}
+svn.revision        120996
+revision            ${svn.revision}
+
+depends_lib-append  port:p5.16-module-depends \
+                    port:p5.16-list-moreutils \
+                    port:p5.16-module-corelist \
+                    port:p5.16-yaml \
+                    port:p5.16-file-path \
+                    port:p5.16-pod-simple \
+                    port:p5.16-carp-clan \
+                    port:p5.16-getopt-simple \
+                    port:p5.16-parse-cpan-meta \
+                    port:p5.16-cpan-meta-yaml
 
 worksrcdir          ${name}
 
-configure.perl      ${prefix}/bin/perl5.12
+configure.perl      ${prefix}/bin/perl5.16
 
 configure {
     reinplace "s|#! /usr/bin/env perl|#!${configure.perl}|g" ${worksrcpath}/${name}
 }
 
-build {}
+# should be implied by overriding configure:
+configure.ccache    no
+
+build {
+    ui_debug "${name} has no build step."
+}
 
 destroot {
     xinstall -m 755 ${worksrcpath}/${name} ${destroot}${prefix}/bin/${name}
