--- magic/Magdir/msdos.orig	2014-06-03 15:17:27.000000000 -0400
+++ magic/Magdir/msdos	2014-10-10 12:42:44.000000000 -0400
@@ -830,8 +830,11 @@
 0	string/b	\224\246\056		Microsoft Word Document
 !:mime	application/msword
 
+## this is not completely accurate, there are Excel documents that have the same pattern
+## 512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	Microsoft Word Document
+## !:mime	application/msword
 512	string	R\0o\0o\0t\0\ \0E\0n\0t\0r\0y	Microsoft Word Document
-!:mime	application/msword
+!:mime	application/x-msoffice
 
 # From: "Nelson A. de Oliveira" <naoliv@gmail.com>
 # Magic type for Dell's BIOS .hdr files
