--- src/Makefile.am.orig	2013-02-22 09:03:35.000000000 -0500
+++ src/Makefile.am	2014-10-10 12:51:17.000000000 -0400
@@ -20,11 +20,11 @@
 libmagic_la_LIBADD = $(LTLIBOBJS) $(MINGWLIBS)
 
 file_SOURCES = file.c
-file_LDADD = libmagic.la
+file_LDADD = libmagic.la $(LTLIBOBJS)
 CLEANFILES = magic.h
 EXTRA_DIST = magic.h.in
 HDR= $(top_srcdir)/src/magic.h.in
 BUILT_SOURCES = magic.h
 
 magic.h:	${HDR}
-	sed -e "s/X.YY/$$(echo @VERSION@ | tr -d .)/" < ${HDR} > $@
+	$(SED) -e "s/X.YY/$$(echo @VERSION@ | tr -d .)/" < ${HDR} > $@
