--- calendar/libedata-cal/e-cal-backend-intervaltree.c.orig	2011-04-21 15:35:36.000000000 -0400
+++ calendar/libedata-cal/e-cal-backend-intervaltree.c	2014-05-15 16:48:24.000000000 -0400
@@ -21,12 +21,28 @@
  */

 #ifdef HAVE_CONFIG_H
-#include <config.h>
-#endif
+# include <config.h>
+#endif /* HAVE_CONFIG_H */

 #include <stdio.h>
 #include <string.h>
-#include <malloc.h>
+#ifndef malloc
+# ifdef HAVE_STDLIB_H
+#  include <stdlib.h>
+# else
+#  ifdef HAVE_MALLOC_H
+#   include <malloc.h>
+#  else
+#   ifdef HAVE_MALLOC_MALLOC_H
+#    include <malloc/malloc.h>
+#   else
+#    if defined(__GNUC__) && !defined(__STRICT_ANSI__)
+#     warning "e-cal-backend-intervaltree.c cannot find a header to include for malloc()."
+#    endif /* __GNUC__ && !__STRICT_ANSI__ */
+#   endif /* HAVE_MALLOC_MALLOC_H */
+#  endif /* HAVE_MALLOC_H */
+# endif /* HAVE_STDLIB_H */
+#endif /* !malloc */

 #include "e-cal-backend-intervaltree.h"

