--- Portfile.orig
+++ Portfile
@@ -5,29 +5,22 @@
 
 name                py-sarif-om
 version             1.0.4
-revision            0
-
-categories          python
+platforms           {darwin any}
 supported_archs     noarch
 license             MIT
-maintainers         {judaew @judaew} openmaintainer
+maintainers         nomaintainer
 
 description         Classes implementing the SARIF 2.1.0 object model.
 long_description    {*}${description}
+
 homepage            https://github.com/microsoft/sarif-python-om
-
-distname            sarif_om-${version}
-
-checksums           rmd160  2040375edf41e8fe6ee793e7bfc6afda91192fb6 \
-                    sha256  cd5f416b3083e00d402a92e449a7ff67af46f11241073eea0461802a3b5aef98 \
+checksums           rmd160  2040375edf41e8fe6ee793e7bfc6afda91192fb6\n \
+                    sha256  cd5f416b3083e00d402a92e449a7ff67af46f11241073eea0461802a3b5aef98\n \
                     size    28847
 
-python.versions     39 310 311
+python.versions     311 >= 2.7
+python.pep517       yes
 
 if {${name} ne ${subport}} {
-    depends_build-append \
-                        port:py${python.version}-setuptools
-    depends_lib-append  port:py${python.version}-pbr
-
     livecheck.type      none
 }
