--- Makefile.am	2013-03-29 11:46:32.000000000 -0400
+++ /Users/ericgallager/GitHubReposFromOtherComputers/scilab/scilab/Makefile.am	2013-09-17 18:16:27.000000000 -0400
@@ -12,13 +12,14 @@
 
 ## Process this file with automake to produce Makefile.in (which will be changed by the ./configure)
 
-AUTOMAKE_OPTIONS = 1.11 dist-zip dist-bzip2
+AUTOMAKE_OPTIONS = 1.11 dejagnu dist-zip dist-bzip2 foreign subdir-objects -Wall
 
-SUBDIRS= modules \
+SUBDIRS = modules \
 	contrib \
-	desktop
+	desktop \
+        po
 
-ACLOCAL_AMFLAGS = -I m4/
+ACLOCAL_AMFLAGS = -I m4
 
 # Set AM_* SCI detected settings
 AM_CPPFLAGS = $(SCI_CPPFLAGS)
@@ -31,28 +32,28 @@
 
 bin_PROGRAMS		= scilab-bin scilab-cli-bin
 
-bin_SCRIPTS			= bin/scilab bin/scilab-adv-cli bin/scilab-cli \
+bin_SCRIPTS		= bin/scilab bin/scilab-adv-cli bin/scilab-cli \
 bin/scinotes bin/xcos
 
 if IS_MACOSX
 bin_SCRIPTS 		+= bin/checkmacosx.applescript
 endif
 
-scilab_bin_LDFLAGS 	    = $(AM_LDFLAGS)
+scilab_bin_LDFLAGS 	= $(AM_LDFLAGS)
 scilab_cli_bin_LDFLAGS  = $(AM_LDFLAGS)
 
 if IS_MACOSX
-scilab_bin_LDFLAGS 		+= -framework Cocoa
+scilab_bin_LDFLAGS 	+= "-Wl,-framework,Cocoa"
 endif
 
 scilab_bin_CPPFLAGS = -I$(top_srcdir)/modules/io/includes/ \
-                    -I$(top_srcdir)/modules/console/includes/ \
-                    -I$(top_srcdir)/modules/localization/includes/ \
-                    $(AM_CPPFLAGS)
+                      -I$(top_srcdir)/modules/console/includes/ \
+                      -I$(top_srcdir)/modules/localization/includes/ \
+                      $(AM_CPPFLAGS)
 # --no-undefined --relax
 
-# Localization dep already exists because of the `exportLocaleToSystem' symbol
-# (causing issues on some platform)
+# Localization dep already exists because of the `exportLocaleToSystem'
+# symbol (causing issues on some platform)
 COMMON_LIBS = $(LTLIBINTL)
 
 scilab_bin_SOURCES		= $(top_srcdir)/modules/console/src/c/others/mainscic.c
@@ -67,7 +68,7 @@
 
 
 if NEED_JAVA
-# Don't do it for scilab_cli_bin. We want to be able to call it without
+# Do not do it for scilab_cli_bin. We want to be able to call it without
 # the java libs
 	scilab_bin_LDADD += $(JAVA_JNI_LIBS)
 endif
@@ -82,15 +83,19 @@
 scilab_static_bin_CFLAGS  = -Wl,--start-group `$(CXX) -print-file-name=libstdc++.a` $(AM_CFLAGS)
 
 #-static
-bin_PROGRAMS	   		+= scilab-static-bin
-scilab_static_bin_LDADD = $(top_builddir)/modules/libscilab.la
+bin_PROGRAMS	   	  += scilab-static-bin
+scilab_static_bin_LDADD   = $(top_builddir)/modules/libscilab.la
 
 scilab_static_bin_SOURCES = $(scilab_bin_SOURCES)
 endif
 
 LANG_DOMAIN=scilab
 
-TARGETS_ALL=macros
+TARGETS_ALL=
+# The user wants to build macros
+if BUILD_MACROS
+TARGETS_ALL+=macros
+endif
 
 #  The user wants to compile the localization files
 if GENERATE_LOCALIZATION_FILES
@@ -102,12 +107,13 @@
 endif
 
 # check is performed with scilab
-TESTS_ENVIRONMENT = LANG=C $(top_builddir)/bin/scilab-cli -nb -noatomsautoload -f
+TESTS_ENVIRONMENT = LANG=C LC_ALL=C # $(top_builddir)/bin/scilab-cli -nb -noatomsautoload -f
 TESTS = $(top_srcdir)/modules/development_tools/tests/unit_tests/test_run_level_1.tst
 
 all-local: $(TARGETS_ALL)
 
-rootdir=$(pkgdatadir)/etc/
+rootdir = $(pkgdatadir)/etc
+
 root_DATA = $(top_srcdir)/etc/scilab.start \
 $(top_srcdir)/etc/scilab.quit \
 $(top_srcdir)/etc/toolboxes.xml \
@@ -117,7 +123,8 @@
 $(top_builddir)/etc/logging.properties \
 $(top_builddir)/etc/librarypath.xml
 
-infofilesdir=$(pkgdatadir)/
+infofilesdir = $(pkgdatadir)
+
 infofiles_DATA = $(top_srcdir)/CHANGES_2.X \
 $(top_srcdir)/CHANGES_3.X \
 $(top_srcdir)/CHANGES_4.X \
@@ -139,7 +146,7 @@
 $(top_srcdir)/README_Windows.txt \
 $(top_builddir)/Version.incl
 
-valgrinddir=$(pkgdatadir)/tools/profiling
+valgrinddir = $(pkgdatadir)/tools/profiling
 valgrind_DATA = $(top_srcdir)/tools/profiling/valgrind.supp
 
 macros: $(top_builddir)/scilab-cli-bin $(top_builddir)/bin/scilab-cli check-jvm-dep
@@ -280,12 +287,14 @@
 
 clean-macros:
 # Find is necessary because of subdirs...
+	rm -f $(top_builddir)/scilab-bin
+	rm -f $(top_builddir)/scilab-cli-bin
 	-@echo "-- Cleaning macros (*.bin) --"; \
-	find $(top_builddir) -name "*.bin" | xargs --no-run-if-empty rm 2> /dev/null; \
+	find $(top_builddir) -name "*.bin" | xargs --no-run-if-empty rm 2>/dev/null; \
 	echo "-- Cleaning names files (names) --"; \
-	find $(top_builddir)/modules/*/macros -name "names" | xargs --no-run-if-empty rm 2> /dev/null; \
+	find $(top_builddir)/modules/*/macros -name "names" | xargs --no-run-if-empty rm 2>/dev/null; \
 	echo "-- Cleaning lib files (lib) --"; \
-	find $(top_builddir)/modules/*/macros -name "lib" | xargs --no-run-if-empty rm 2> /dev/null
+	find $(top_builddir)/modules/*/macros -name "lib" | xargs --no-run-if-empty rm 2>/dev/null
 
 # To ensure previous compatibility
 clean-doc: clean-help
@@ -322,22 +331,29 @@
 
 clean-localization:
 	rm -rf $(top_srcdir)/locale/*
+	rm -f $(top_builddir)/scilab-bin
+	rm -f $(top_builddir)/scilab-cli-bin
 
 clean-generated:
 # Deleted junit + code coverage
-	rm -rf $(top_builddir)/reports/
-	rm -rf $(top_builddir)/modules/scicos_blocks/build/ 2> /dev/null
+	rm -f $(top_builddir)/scilab-bin
+	rm -f $(top_builddir)/scilab-cli-bin
+	rm -rf $(top_builddir)/reports
+	rm -rf $(top_builddir)/modules/scicos_blocks/build 2>/dev/null
 
 distclean-generated:
-	rm -f $(top_builddir)/modules/core/includes/stack.h 2> /dev/null
+	rm -f $(top_builddir)/modules/core/includes/stack.h 2>/dev/null
 
 
 if IS_MACOSX
-# Used for mac os x because it needs to remove the charset=UTF-8 because it is breaking the accents.
+# Used for Mac OS X because it needs to remove the charset=UTF-8
+# because otherwise it breaks the accents.
 MACOSX=1
 endif
 
 clean-reports:
+	rm -f $(top_builddir)/scilab-bin
+	rm -f $(top_builddir)/scilab-cli-bin
 	rm -f reports/xml/TEST-org.scilab.*.xml
 
 # For each module, build and call Java unitary tests
