--- lib/dpkg/tarfn.c.orig	2013-09-30 10:47:56.000000000 -0400
+++ lib/dpkg/tarfn.c	2014-03-27 10:24:59.000000000 -0400
@@ -40,6 +40,9 @@
 #define TAR_MAGIC_USTAR "ustar\0" "00"
 #define TAR_MAGIC_GNU   "ustar "  " \0"
 
+static const char ustarMagic[] = { 'u', 's', 't', 'a', 'r', '\0', '0', '0', '\0' };
+static const char gnutarMagic[] = { 'u', 's', 't', 'a', 'r', ' ', ' ', '\0' };
+
 struct tar_header {
 	char name[100];
 	char mode[8];
@@ -51,6 +54,7 @@
 	char linkflag;
 	char linkname[100];
 	char magic[8];
+    char MagicNumber[8]; /* POSIX: "ustar\000", GNU: "ustar  \0" (blank blank null) */
 	char user[32];
 	char group[32];
 	char devmajor[8];
@@ -58,6 +62,7 @@
 
 	/* Only valid on ustar. */
 	char prefix[155];
+    char Prefix[155]; /* POSIX ustar header */
 };
 
 /**
